src/OpenApi/src/Comparers/OpenApiDiscriminatorComparer.cs

```diff
@@ -27,7 +26,7 @@ public bool Equals(OpenApiDiscriminator? x, OpenApiDiscriminator? y)
 
         return x.PropertyName == y.PropertyName &&
             x.Mapping.Count == y.Mapping.Count &&
-            x.Mapping.Keys.All(key => y.Mapping.TryGetValue(key, out var yValue) && x.Mapping[key] == yValue);
+            ComparerHelpers.DictionaryEquals((Dictionary<string, string>)x.Mapping, (Dictionary<string, string>)y.Mapping, StringComparer.Ordinal);
```

Comment:
I don't think these should be casted without checking the type. OpenAPI library automatically creates Dictionary and List instances for these properties, but someone could set their own IDictionary implementation.

Reply:
Yep, I shared this insight [over in the other PR](https://github.com/dotnet/aspnetcore/pull/56599#discussion_r1671411261). I was optimizing for code brevity here given how unlikely it is for these to be overridden with non-default implementations but I can adjust if we want things to be a little sturdies here.

Reply:
I think making ListEquals and DictionaryEquals a little bigger to handle other types is worth the safety. Also, it allows the casts to be removed when calling them.